test/e2e/End2EndSpec.scala

```diff
@@ -43,4 +47,50 @@ class End2EndSpec(arguments: Arguments) extends Specification with GuiceFakeAppl
     customArgumentsMap.groupBy(_(0).substring(2)).view.mapValues(_(0).last).toMap
   }
 
+  private def ensureTestDataset(): Unit = {
+    val testDatasetPath = "test/dataset/test-dataset.zip"
+    val dataDirectory = new File("binaryData/Organization_X")
+    if (!dataDirectory.exists()) {
+      throw new Exception("Organization_X directory does not exist.")
+    }
+    val testDatasetZip = new File(testDatasetPath)
+    if (!testDatasetZip.exists()) {
+      throw new Exception("Test dataset zip file does not exist.")
+    }
+    // Stop if the test dataset is already unzipped
+    if (dataDirectory.listFiles().exists(_.getName == "test-dataset")) {
+      return
+    }
+    unzipArchive(testDatasetPath, dataDirectory)
+  }
+
+  private def unzipArchive(archivePath: String, destination: File): Unit = {
+    val buffer = new Array[Byte](1024)
+    val zis = new ZipInputStream(new FileInputStream(archivePath))
+    var zipEntry = zis.getNextEntry
+    while (zipEntry != null) {
+      val newFile = new File(destination, zipEntry.getName)
+      if (zipEntry.isDirectory) {
+        if (!newFile.isDirectory && !newFile.mkdirs) {
+          throw new IOException("Failed to create directory " + newFile)
+        }
+      } else {
+        val parent = newFile.getParentFile
+        if (!parent.isDirectory && !parent.mkdirs)
+          throw new IOException("Failed to create directory " + parent)
+        val fos = new FileOutputStream(newFile)
+        var len = zis.read(buffer)
+        while (len > 0) {
+          fos.write(buffer, 0, len)
+          len = zis.read(buffer)
+        }
+        fos.close()
+
+      }
+      zipEntry = zis.getNextEntry
+    }
+
+    zis.closeEntry()
```

Comment:
This seems quite involved. Is there a reason this does not use the ZipIO.unzipToFolder function that we already have?

Reply:
There were no dependencies to any other local package in the test package and I thought that may have been intentional. I have no reason in particular.

Reply:
I see, well I don’t know if that was intended originally, but to me using the zip utils here seems acceptable. We might even catch bugs there if this test fails…

Reply:
Changed it